src/Features/header/components/CityLanguageSelect/City/CitySelect.tsx

```diff
@@ -0,0 +1,81 @@
+import Link from 'next/link';
+import React, { ReactElement, useState, useEffect } from 'react';
+import { setCookie, getCookie, CookieValueTypes } from 'cookies-next';
+import { Modal } from '../../../../../components/Modal/Modal';
+
+interface CitySelectProps {
+  myValue: string | undefined;
+}
+
+export const CitySelect: React.FC<CitySelectProps> = ({ myValue }): ReactElement => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [selectedValue, setSelectedValue] = useState<string | undefined>(myValue);
+
+  useEffect(() => {
+    const cookieValue = getCookie('myValue') as CookieValueTypes;
+    if (typeof cookieValue === 'string') {
+      setSelectedValue(cookieValue || undefined);
+    }
+  }, []);
+
+  const handleOpen = (event: React.MouseEvent<HTMLElement>) => {
+    event.preventDefault();
+    setShowModal(true);
+  };
+
+  const selectValue = (value: string) => {
+    const expires = new Date();
```

Comment:
It is cool, but I have 2 suggestions: 1) create it in shorter form like `export const SEVEN_DAYS = new Date(new Date().getTime() + 7 * 24 * 60 * 60 * 1000)` 2) put in the constants on the `src/constants` level. It might reused later